cmd/contour/serve.go

```diff
@@ -389,16 +389,22 @@ func (s *Server) doServe() error {
 		return err
 	}
 
+	var applyHeaderPolicyToIngress bool
+	if contourConfiguration.Policy != nil {
+		applyHeaderPolicyToIngress = contourConfiguration.Policy.ApplyToIngress
+	}
+
 	builder := s.getDAGBuilder(dagBuilderConfig{
-		ingressClassName:          ingressClassName,
-		rootNamespaces:            contourConfiguration.HTTPProxy.RootNamespaces,
-		gatewayAPIConfigured:      contourConfiguration.Gateway != nil,
-		disablePermitInsecure:     contourConfiguration.HTTPProxy.DisablePermitInsecure,
-		enableExternalNameService: contourConfiguration.EnableExternalNameService,
-		dnsLookupFamily:           contourConfiguration.Envoy.Cluster.DNSLookupFamily,
-		headersPolicy:             contourConfiguration.Policy,
-		clientCert:                clientCert,
-		fallbackCert:              fallbackCert,
+		ingressClassName:           ingressClassName,
+		rootNamespaces:             contourConfiguration.HTTPProxy.RootNamespaces,
+		gatewayAPIConfigured:       contourConfiguration.Gateway != nil,
+		disablePermitInsecure:      contourConfiguration.HTTPProxy.DisablePermitInsecure,
+		enableExternalNameService:  contourConfiguration.EnableExternalNameService,
+		dnsLookupFamily:            contourConfiguration.Envoy.Cluster.DNSLookupFamily,
+		headersPolicy:              contourConfiguration.Policy,
```

Comment:
nit: since we pass in the whole headers policy object, could we just grab it from there instead of a separate field in the dag builder config? same difference but one less field to fill in

Reply:
Yep makes sense